android/tests_backend/widgets/base.py

```diff
@@ -231,6 +231,12 @@ def is_hidden(self):
     def has_focus(self):
         return self.widget.app._impl.native.getCurrentFocus() == self.native
 
+    async def undo(self):
+        raise NotImplementedError()
```

Comment:
Right idea, slightly wrong implementation. If you make the body of this method `pytest.skip()`, then the test will be skipped on Android as soon as the test tries to invoke it, removing the need to explicitly skip the platforms in the test itself. See [canvas on Android](https://github.com/beeware/toga/blob/main/android/tests_backend/widgets/canvas.py#L39) for an example.